syntax = "proto3";

package rpc;

service FaceIdAi {
	rpc GetFeatureVector(Image) returns (FaceEmbedding) {}
	rpc BeAtYourDesk(EmployeeInfo) returns (EmployeeStatus) {}
}

message EmployeeInfo {
	uint64 id = 1;
	Image image = 2;
}

message EmployeeStatus {
	bool status = 1;
}

message Image {
	bytes encoded_image = 1;
}

message FaceEmbedding {
	FeatureVector feature_vector = 1;
	enum State {
		STATE_UNSPECIFIED = 0;
		STATE_SUCCESS = 1;
		STATE_NOT_FOUND = 2;
		STATE_MULTIFACE = 3;
		STATE_ERROR = 4;				// Can be further expanded into error types
	}
	State vector_state = 2;
}

message FeatureVector {
	repeated float vector = 1;
}
